Description: Include <ctime> for all cpp files needed to avoid
building failed.
Author: Xiangwei Gui <yuemeng.gui@gmail.com>
--- a/src/examples/adhoc_example.cpp
+++ b/src/examples/adhoc_example.cpp
@@ -25,6 +25,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 
 class AdhocTest : public ConnectionListener, AdhocCommandProvider, LogHandler
--- a/src/examples/annotations_example.cpp
+++ b/src/examples/annotations_example.cpp
@@ -22,6 +22,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class AnnotationsTest : public AnnotationsHandler, ConnectionListener
 {
--- a/src/examples/bookmarkstorage_example.cpp
+++ b/src/examples/bookmarkstorage_example.cpp
@@ -22,6 +22,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class BookmarkStorageTest : public BookmarkHandler, ConnectionListener
 {
--- a/src/examples/component_example.cpp
+++ b/src/examples/component_example.cpp
@@ -22,6 +22,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class ComponentTest : public DiscoHandler, ConnectionListener, LogHandler
 {
--- a/src/examples/disco_example.cpp
+++ b/src/examples/disco_example.cpp
@@ -24,6 +24,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class DiscoTest : public DiscoHandler, ConnectionListener, LogHandler
 {
--- a/src/examples/e2ee_client.cpp
+++ b/src/examples/e2ee_client.cpp
@@ -29,6 +29,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #ifdef WIN32
 #include <windows.h>
--- a/src/examples/e2ee_server.cpp
+++ b/src/examples/e2ee_server.cpp
@@ -29,6 +29,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #ifdef WIN32
 #include <windows.h>
--- a/src/examples/flexoff_example.cpp
+++ b/src/examples/flexoff_example.cpp
@@ -28,6 +28,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class FlexOffTest : public MessageHandler, ConnectionListener, FlexibleOfflineHandler,
                            LogHandler
--- a/src/examples/ft_recv.cpp
+++ b/src/examples/ft_recv.cpp
@@ -27,6 +27,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #if defined( WIN32 ) || defined( _WIN32 )
 # include <windows.h>
--- a/src/examples/privacylist_example.cpp
+++ b/src/examples/privacylist_example.cpp
@@ -22,6 +22,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class PLTest : public PrivacyListHandler, ConnectionListener
 {
--- a/src/examples/privatexml_example.cpp
+++ b/src/examples/privatexml_example.cpp
@@ -22,6 +22,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class PrivateXMLTest : public PrivateXMLHandler, ConnectionListener
 {
--- a/src/examples/register_example.cpp
+++ b/src/examples/register_example.cpp
@@ -22,6 +22,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class RegTest : public RegistrationHandler, ConnectionListener, LogHandler
 {
--- a/src/examples/reset_example.cpp
+++ b/src/examples/reset_example.cpp
@@ -23,6 +23,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class RosterTest : public ConnectionListener, LogHandler
 {
--- a/src/examples/roster_example.cpp
+++ b/src/examples/roster_example.cpp
@@ -27,6 +27,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class RosterTest : public RosterListener, ConnectionListener, LogHandler, MessageHandler
 {
--- a/src/examples/vcard_example.cpp
+++ b/src/examples/vcard_example.cpp
@@ -26,6 +26,7 @@
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 class VCardTest : public ConnectionListener, LogHandler, VCardHandler
 {
--- a/src/tests/tag/tag_perf.cpp
+++ b/src/tests/tag/tag_perf.cpp
@@ -22,6 +22,7 @@
 #include <cstdio> // [s]print[f]
 
 #include <sys/time.h>
+#include <ctime>
 
 static double divider = 1000000;
 static int num = 2500;
--- a/src/tests/zlib/zlib_perf.cpp
+++ b/src/tests/zlib/zlib_perf.cpp
@@ -24,6 +24,7 @@
 #include <string>
 #include <sys/time.h>
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #ifdef HAVE_ZLIB
 
